js/src/components/tree-select-control/index.js

```diff
@@ -147,14 +147,14 @@ const TreeSelectControl = ( {
 					return;
 				}
 
-				const childIdPosition = newValue.indexOf( child.id );
+				const childIdPosition = newValue.indexOf( child.value );
```

Comment:
❔ Do we want to rename this `childIdPosition`, because it is not really "id" anymore? Maybe just something simple like `idx` or `index`?

Reply:
Good suggestion @ecgan 

Changed here: https://github.com/woocommerce/google-listings-and-ads/pull/1279/commits/0214f34487ea25ebc044d1c5d1ebde5cb7ad029c